packages/zowe-explorer/src/commands/TsoCommandHandler.ts

```diff
@@ -72,52 +84,27 @@ export class TsoCommandHandler extends ZoweCommandProvider {
-            const profileNamesList = ProfileManagement.getRegisteredProfileNameList(Definitions.Trees.MVS);
-            if (profileNamesList.length > 0) {
-                const quickPickOptions: vscode.QuickPickOptions = {
-                    placeHolder: vscode.l10n.t("Select the Profile to use to submit the TSO command"),
-                    ignoreFocusOut: true,
-                    canPickMany: false,
-                };
-                const sesName = await Gui.showQuickPick(profileNamesList, quickPickOptions);
-                if (sesName === undefined) {
-                    Gui.showMessage(vscode.l10n.t("Operation Cancelled"));
-                    return;
-                }
-                const allProfiles = profiles.allProfiles;
-                profile = allProfiles.filter((temprofile) => temprofile.name === sesName)[0];
-                if (!node) {
-                    await profiles.checkCurrentProfile(profile);
-                }
-                if (profiles.validProfile !== Validation.ValidationType.INVALID) {
-                    session = ZoweExplorerApiRegister.getMvsApi(profile).getSession();
-                } else {
-                    Gui.errorMessage(vscode.l10n.t("Profile is invalid"));
-                    return;
-                }
-            } else {
-                Gui.showMessage(vscode.l10n.t("No profiles available"));
+            profile = await this.selectNodeProfile(Definitions.Trees.MVS);
+            if (!profile) {
                 return;
             }
         } else {
             profile = node.getProfile();
         }
         try {
-            if (profiles.validProfile !== Validation.ValidationType.INVALID) {
+            if (this.profileInstance.validProfile !== Validation.ValidationType.INVALID) {
                 const commandApi = ZoweExplorerApiRegister.getInstance().getCommandApi(profile);
                 if (commandApi) {
-                    let tsoParams: zostso.IStartTsoParms;
                     if (profile.type === "zosmf") {
-                        tsoParams = await this.getTsoParams();
-                        if (!tsoParams) {
+                        this.tsoParams = await this.getTsoParams();
+                        if (!this.tsoParams) {
                             return;
                         }
                     }
-                    let command1: string = command;
                     if (!command) {
-                        command1 = await this.getQuickPick(session && session.ISession ? session.ISession.hostname : "unknown");
+                        command = await this.getQuickPick([session && session.ISession ? session.ISession.hostname : "unknown"]);
                     }
-                    await this.issueCommand(command1, profile, tsoParams);
+                    await this.issueCommand(profile, command);
                 } else {
                     Gui.errorMessage(vscode.l10n.t("Profile is invalid"));
                     return;
```

Comment:
I guess I have the same question for the other command handlers as well, but I see that this pattern has existed in these command handlers for some time 😅

Reply:
see above comment